nums/core/array/blockarray.py

```diff
@@ -883,21 +885,41 @@ def _block_shape_reshape(self, arr, block_shape):
     def _strip_ones(self, shape):
         return tuple(filter(lambda x: x != 1, shape))
 
-    def _is_simple_reshape(self, arr: BlockArray, shape, block_shape):
+    def _is_simple_reshape(self, arr: BlockArray, shape):
         # Is the reshape a difference of factors of 1?
         # Strip out 1s and compare.
-        return self._strip_ones(shape) == self._strip_ones(
-            arr.shape
-        ) and self._strip_ones(block_shape) == self._strip_ones(arr.block_shape)
+        return self._strip_ones(shape) == self._strip_ones(arr.shape)
 
     def _simple_reshape(self, arr, shape, block_shape):
         # Reshape the array of blocks only.
         # This is only used when the difference in shape are factors of 1s,
         # and the ordering of other factors are maintained.
 
-        # Check assumptions.
-        assert len(self._strip_ones(arr.shape)) == len(self._strip_ones(shape))
+        # Strip ones
+        arr_shape_stripped = self._strip_ones(arr.shape)
+        new_shape_stripped = self._strip_ones(shape)
 
+        # Check assumptions.
+        assert len(arr_shape_stripped) == len(new_shape_stripped)
+
+        if block_shape == shape:
+            indices = range(len(shape))
+            new_block_shape = []
+            arr_shape = list(arr.shape)
+            for i in indices:
+                if shape[i] in arr_shape_stripped:
+                    index = arr_shape.index(shape[i])
+                    new_block_shape.append(arr.block_shape[index])
+                    arr_shape[index] = -1
+                else:
+                    new_block_shape.append(1)
+            block_shape = tuple(new_block_shape)
+        else:
+            new_grid_shape = map(lambda i, j: (-(-i // j)), shape, block_shape)
+            grid_shape_list = list(list(new_grid_shape))
+            product = functools.reduce(lambda a, b: a * b, grid_shape_list)
+            if product != np.product(arr.grid.grid_shape):
+                raise ValueError("Incompatible block shape")
```

Comment:
Could you describe the purpose of this test? If `shape = (1,2,3)`, `block_shape=(1,2,3)` and `arr.shape = (2,3)`, `arr.block_shape = (2,2)`, then we have `grid_shape_a = (1, 1, 1)` and `grid_shape_b = (1, 2)`, and their products will not be equal, but the reshape should still be possible via a more general routine. Is there something I'm missing?